packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2700,7 +2700,10 @@ export class ContainerRuntime
 				this.updateDocumentDirtyState(false);
 			}
 
-			this.validateAndProcessRuntimeMessage(messageWithContext, localOpMetadata);
+			// validateAndProcessRuntimeMessage only processes internal container messages
+			if (messageWithContext.message.type in ContainerMessageType) {
```

Comment:
I am wondering if it should skip everything except for emitting the "op".

Here's one unintended consequence:
Messages that are generated by server such as join, leave, summary acks, etc. will have clientSequenceNumber set to -1. So, the code above `this._processedClientSequenceNumber = message.clientSequenceNumber` will set `_processedClientSequenceNumber` to -1 which is incorrect. Basically, the client sequence number goes back in time.
@kian-thompson client sequence number is used for op grouping or for other op processing right? Will this cause issues?

Reply:
There is logic for before/after scheduleManager processing, resetReconnectCount logic as well, which I thought were relevant to all ops. But I also don't have all the context, so let me know if that is not true. I also try to spend some more time looking at code/docs to confirm.

Reply:
All those logics should only care about messages sent by clients (and not server). However, I am not sure if that would break anything. So, I would suggest skipping `validateAndProcessRuntimeMessage` and `this._processedClientSequenceNumber = message.clientSequenceNumber` for now. The other will require more investigation.
Also, maybe wait for Kian to look into what affect (if any) updating client sequence number to -1 has.

Reply:
> Basically, the client sequence number goes back in time

I wouldn't say "goes back in time", in practice the only purpose for `_processedClientSequenceNumber` is to detect if it changes while a batch is being built up, to ensure that nothing comes in and interrupts the batch (e.g. during the gap when waiting for async flush).

The code that checks this also checks `referenceSequenceNumber`.  Before Grouped Batching, the CSN check was not needed - If we processed another op, ref seq would increment.  But after Grouped Batching hit, @kian-thompson did PR #15088 (esp https://github.com/microsoft/FluidFramework/pull/15088/commits/96ab0d5b36df5a50fcd30e496d0a155dfa4422f3) which also added CSN check.  Because as we progress through processing a Grouped Batch, ref seq doesn't advance, but CSN does (because it's actually "index into batch" at this point in that case).

So what does this mean for non-modern-runtime-messages?  Grouped Batching doesn't apply.  So the ref seq check will be sufficient.  So we should be fine to omit this.

Of course, I could have something wrong here, so there's some risk.

Reply:
PS (beyond the scope of this PR) - @kian-thompson, @vladsud -- How could only part of a grouped batch interrupt a batch being built up?? Batches are process synchronously.  So I think the only case would be if the Grouped Batch messages trigger op reentrancy multiple times, such that the reentrant ops are interleaved with the incoming batch messages.

But in that case, we could just let the batch build up and then when it's flushed it will rebase it.  This seems better than submitting the ops one at a time which is what will happen (via `Outbox.maybeFlushPartialBatch`).

i.e. I'm proposing:  Remove the tracking of CSN for grouped batches, go back to just monitoring changes in Ref Seq.